frontend/src/test/scala/bloop/dap/DebugSessionProxy.scala

```diff
@@ -0,0 +1,107 @@
+package bloop.dap
+
+import java.net.{Socket, SocketException}
+import java.nio.ByteBuffer
+
+import bloop.bsp.BloopLanguageClient
+import com.microsoft.java.debug.core.protocol.Messages.ProtocolMessage
+import com.microsoft.java.debug.core.protocol.{JsonUtils, Messages}
+import monix.eval.Task
+import monix.execution.{Ack, Scheduler}
+import monix.reactive.observables.ObservableLike.Operator
+import monix.reactive.observers.Subscriber
+import monix.reactive.{Observable, Observer}
+
+import scala.collection.mutable
+import scala.concurrent.{Future, Promise}
+import scala.meta.jsonrpc.{BaseProtocolMessage, MessageWriter}
+
+private[dap] final class DebugSessionProxy(
```

Comment:
This looks correct, but could you explain a little bit more how all of this testing infrastructure works at a high-level with in-depth Scaladocs?

Reply:
I have given it a go, but I am not very proficient at creating good docs. Could you (or @tgodzik ) take a look and point me in the right direction (what to update etc.)?